landa/landa_sales/party.py

```diff
@@ -0,0 +1,115 @@
+import frappe
+from erpnext.accounts.party import get_party_details
+from frappe.contacts.doctype.address.address import (
+	get_address_display,
+)
+from frappe.contacts.doctype.contact.contact import get_contact_details
+
+EMPTY_CONTACT = {
+	"contact_person": None,
+	"contact_display": None,
+	"contact_email": None,
+	"contact_mobile": None,
+	"contact_phone": None,
+}
+
+
+@frappe.whitelist()
+def get_landa_party_details(
+	party=None,
+	account=None,
+	party_type="Customer",
+	company=None,
+	posting_date=None,
+	bill_date=None,
+	price_list=None,
+	currency=None,
+	doctype=None,
+	ignore_permissions=False,
+	fetch_payment_terms_template=True,
+	party_address=None,
+	company_address=None,
+	shipping_address=None,
+	pos_profile=None,
+):
+	party_details = get_party_details(
+		party,
+		account,
+		party_type,
+		company,
+		posting_date,
+		bill_date,
+		price_list,
+		currency,
+		doctype,
+		ignore_permissions,
+		fetch_payment_terms_template,
+		party_address,
+		company_address,
+		shipping_address,
+		pos_profile,
+	)
+
+	if party_type == "Customer" and party:
+		customer = frappe.get_doc("Customer", party)
+
+		set_billing_address(party_details, customer)
+		set_billing_contact(party_details, customer, doctype)
+		set_shipping_address(party_details, customer)
+		set_shipping_contact(party_details, customer, doctype)
+
+	return party_details
+
+
+def set_billing_address(party_details, customer):
+	party_details["customer_address"] = customer.default_billing_address
+	party_details["address_display"] = get_address_display(
+		customer.default_billing_address
+	)
+
+
+def set_billing_contact(party_details, customer, doctype):
+	if customer.default_billing_contact:
+		billing_contact = get_contact_details(customer.default_billing_contact)
+	else:
+		billing_contact = EMPTY_CONTACT
```

Comment:
```suggestion
		billing_contact = {}
```
We do not need a global variable here. Wherever `billing_contact` is used, we must make sure `get` is used to access keys (which is already done and already returns `None` if the key is absent)

Reply:
If I remember right there were cases where `party_details` contained (incorrect) data from the original `get_party_details` method. - Without `EMPTY_CONTACT` this data won't be erased.